DASH-IF-IOP.bs.md

```diff
@@ -10,8 +10,500 @@ Live discussion in #document-authoring on Slack.
+
+If [=indexed addressing=] is used, `SegmentTemplate@presentationTimeOffset` SHALL be a point within or at the start of any [=media segment=] referenced by the [=period=].
+
+## Media segments ## {#timing-mediasegment}
+
+A <dfn>media segment</dfn> is an HTTP-addressable data structure that contains one or more media samples.
+
+Note: [[MPEGDASH]] makes a distinction between "segment" (HTTP-addressable entity) and "subsegment" (byte range of an HTTP-addressable entity). This document refers to both concepts as "segment".
+
+Issue: Moving to CMAF terms might simplify the linguistic overhead caused by DASH terminology here. See https://github.com/Dash-Industry-Forum/DASH-IF-IOP/issues/217
+
+[=Media segments=] SHALL contain one or more consecutive media samples. Consecutive [=media segments=] in the same [=representation=] SHALL contain consecutive media samples.
+
+[=Media segments=] SHALL contain the media samples that exactly match the time span on the [=sample timeline=] that is mapped to the segment's associated time span on the [=MPD timeline=], except when using [=simple addressing=] in which case a certain amount of inaccuracy MAY be present as defined in [[#timing-addressing-inaccuracy]].
+
+The [=media segment=] that starts at or overlaps the [=period=] start point on the [=MPD timeline=] SHALL contain a media sample that starts at or overlaps the [=period=] start point.
+
+The [=media segment=] that ends at or overlaps the [=period=] end point on the [=MPD timeline=] SHALL contain a media sample that ends at or overlaps the [=period=] end point.
+
+Note: The requirements on providing samples for the [=period=] start/end point in the first/last [=media segment=] apply even when [[#timing-addressing-inaccuracy|inaccurate addressing]] is used.
+
+## Segment addressing modes ## {#timing-addressing}
+
+Issue: Need to go over segment-related terminology and ideally move to CMAF terms (https://github.com/Dash-Industry-Forum/DASH-IF-IOP/issues/217). Right now this is a bit too "media segment" oriented.
+
+This section defines the <dfn title="addressing mode">addressing modes</dfn> that can be used for referencing [=media segments=] in interopreable DASH presentations.
+
+The choice of addressing mode depends on the implementation details of a particular content processing flow. Use whichever addressing mode is suitable for your needs when authoring content.
+
+Addressing modes not listed in this chapter SHALL NOT be used.
+
+All [=representations=] in the same [=adaptation set=] SHALL use the same addressing mode. [=Representations=] in different [=adaptation sets=] MAY use different addressing modes.
+
+### Indexed addressing ### {#timing-addressing-indexed}
+
+Placeholder chapter.
+
+<dfn>indexed addressing</dfn> means `SegmentTemplate` with `@indexRange`.
```

Comment:
I took this from the on-demand profile section in v4.3, which says "SegmentTemplate" (a subtype of SegmentBase). I infer that this means that IOP currently expects SegmentTemplate to be used for on-demand profile addressing. Does this make sense?

Yes, the idea would be to also cover initialization segment matters here.

Reply:
To me, use of `SegmentTemplate` doesn't really make sense when using indexed addressing, because all the (sub)segments are going to be accessed using byte ranges into same URL, meaning the ability to specify a template isn't useful. Also, in the DASH spec it looks like `SegmentTemplate` includes `MultipleSegmentBaseInformation`, the presence of which doesn't make sense when using indexed addressing. It may well be possible to still use `SegmentTemplate` and just omit everything that doesn't make sense (I didn't check), but it feels a bit weird. So I'd expect `SegmentBase` to be used for on-demand.

It's probably worth asking others who were more involved with writing the guidelines up to this point as to what the intention was; my point of view doesn't include any knowledge of discussions that may have already been had on this!

For what it's worth, it looks like existing test vectors are using `SegmentBase`:
https://testassets.dashif.org/#testvector/details/586fb3879ae9045678eaccb4

Reply:
I will raise it separately here to clarify. Perhaps there is an error in IOP v4.3: https://github.com/Dash-Industry-Forum/DASH-IF-IOP/issues/232

Reply:
I assume there is an error in v4.3. Changed to SegmentBase.